### ISSUE TITLE
Sometimes RAIL IR finder does not terminate

### ISSUE THREAD

**Issue by chiniforooshan** [catapult-project/catapult#1486]
The ToT trace viewer hangs when importing this trace file: https://drive.google.com/a/google.com/file/d/0Bz3HKuYDrDnyclNwRFNvQTg0YjQ/view?usp=sharing

It's been made by manually selecting all categories + v8.cpu_profile. Looks like findAllInteractionRecords hangs...

**Comment by benshayden** [comment]
How did you narrow it down to the finder?
How do you know that it doesn't terminate, rather than just being slower than you expect?
That trace is 151MB gunzip'd!
RAILIRFinder was not built for speed, and definitely not for such huge traces. It iterates over the entire model several times.
I'm not sure how to prioritize this. I'd definitely prioritize it higher if there were a trace from deep reports that caused similar issues.

**Comment by chiniforooshan** [comment]
Yeah, I'm not sure how often and when this happens; so didn't add it to the alpha milestone. I still wanted to create a bug with a trace to not loose it.

> How did you narrow it down to the finder?

I just added some logging and narrowed it down to the try block that calls findAllInteractionRecords.

> How do you know that it doesn't terminate, rather than just being slower than you expect?

I don't. But, if I wait long enough the trace viewer crashes.

> That trace is 151MB gunzip'd! RAILIRFinder was not built for speed, and definitely not for such huge traces. It iterates over the entire model several times.

It could be the size. Personally, I think we should eventually be able to process large traces, too, maybe after #627. This was generated based on only 30-45 seconds of interaction; some pages don't even completely load in that time :). However, in this case, I'm not sure if the main issue is the size: I don't have a problem viewing traces in measurmt-traces/desktop-v8-samples which are as large (or even larger).

**Comment by natduca** [comment]
@benshayden can you triage?

**Comment by benshayden** [comment]
Looking.

**Comment by benshayden** [comment]
That trace is still super slow to import, but it had not even gotten to finding RAIL IRs before it crashed when I tried to import it just now.

**Comment by eakuefner** [comment]
@chiniforooshan Would this have been fixed by the streaming stuff? Safe to close?
